File: Trabajos_previos/trabajo_previo_2/sesion_1/sobrecarga_operadores.cpp
#include <iostream>
using namespace std;

// Definicion de la clase Fraccion
class Fraccion {
    int numerador, denominador;

public:
    // Constructor de la clase Fraccion
    Fraccion(int n, int d) : numerador(n), denominador(d) {}

    // Sobrecarga del operador '+'
    Fraccion operator+(const Fraccion &f) {
        // Creación de un objeto Fraccion
        Fraccion resultado(
            numerador * f.denominador + f.numerador,
            denominador * f.denominador
        );
        return resultado;
    }

    void imprimir() {
        cout << numerador << "/" << denominador << endl;
    }
};

int main() {
    Fraccion f1(1, 2);
    Fraccion f2(3, 4);

    // Uso de la sobrecarga del operador '+' para sumar las dos fracciones
    Fraccion f3 = f1 + f2;

    f3.imprimir();

    return 0;
}

File: Trabajos_previos/trabajo_previo_4/sesion_3/sobrecarga_operadores.py
class Point:
    # Constructor
    def __init__(self,x=0,y=0):
        self.x = x
        self.y = y
    
    # Metodo para imprimir
    def __str__(self):
        return "({0},{1})".format(self.x, self.y)
    
    # Metodo para sumar
    def __add__(self, other):
        x = self.x + other.x
        y = self.y + other.y
        return Point(x, y)
    
# creacion de objeto y suma
p1 = Point(1,2)
p2 = Point(2,3)

print(p1+p2)